client/components/wcc-settings-form/settings-item.js

```diff
@@ -23,18 +22,19 @@ const SettingsItem = ( {
 } ) => {
 	const id = layout.key ? layout.key : layout;
 	const updateValue = ( value ) => settingsActions.updateSettingsField( id, value );
-	const updateSubSubValue = ( key, subKey, val ) => settingsActions.updateSettingsObjectSubField( id, key, subKey, val );
-	const removeArrayItem = ( idx ) => settingsActions.removeSettingsArrayFieldItem( id, idx );
+	const updateSubValue = ( key, val ) => settingsActions.updateSettingsField( [ id, key ], val );
+	const removeArrayItem = ( idx ) => settingsActions.removeSettingsField( [ id, idx ] );
 	const savePackage = ( packageData ) => packagesActions.savePackage( id, packageData );
 	const fieldRequired = ( -1 !== schema.required.indexOf( id ) );
-	const fieldValue = settings[ id ];
+	const fieldValue = form.settings[ id ];
 	const fieldSchema = schema.properties[ id ];
+	const fieldType = layout.type || fieldSchema.type || '';
```

Comment:
The type should always come from the `layout`, the `type` field in the layout is mandatory (or it should be).